File: backend/src/models/Route.js
const mongoose = require('mongoose');

const routeSchema = new mongoose.Schema({
    name: {
        type: String,
        required: true,
        trim: true
    },
    code: {
        type: String,
        required: true,
        unique: true,
        uppercase: true,
        trim: true
    },
    type: {
        type: String,
        enum: ['METRO', 'BUS'],
        required: true
    },
    stations: [{
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Station'
    }],
    active: {
        type: Boolean,
        default: true
    },
    description: {
        type: String,
        trim: true
    }
}, {
    timestamps: true
});

// Index for faster queries
routeSchema.index({ code: 1 });
routeSchema.index({ type: 1, active: 1 });

module.exports = mongoose.model('Route', routeSchema);

File: metro_tracker_app/lib/features/metro_tracking/domain/entities/route.dart
import 'package:equatable/equatable.dart';
import 'station.dart';

class MetroRoute extends Equatable {
  final String id;
  final String name;
  final String code;
  final String type;
  final String? description;
  final List<Station> stations;

  const MetroRoute({
    required this.id,
    required this.name,
    required this.code,
    required this.type,
    this.description,
    required this.stations,
  });

  @override
  List<Object?> get props => [id, name, code, type, description, stations];
}
